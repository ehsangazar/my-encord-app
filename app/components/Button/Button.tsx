interface ButtonProps {
  children: React.ReactNode | string;
  colorScheme?: "red" | "blue" | "gray";
  onClick?: () => void;
  type?: "button" | "submit" | "reset";
  disabled?: boolean;
}

const Button = ({ children, colorScheme, ...props }: ButtonProps) => {
  if (colorScheme === "gray") {
    return (
      <button
        className="bg-gray-500 text-white px-4 py-2 rounded-md transition-all duration-300 ease-in-out hover:bg-gray-600 focus:outline-none focus:ring-2 focus:ring-gray-500 focus:ring-opacity-50 w-full disabled:opacity-50 disabled:cursor-not-allowed"
        {...props}
      >
        {children}
      </button>
    );
  }

  if (colorScheme === "red") {
    return (
      <button
        className="bg-red-500 text-white px-4 py-2 rounded-md transition-all duration-300 ease-in-out hover:bg-red-600 focus:outline-none focus:ring-2 focus:ring-red-500 focus:ring-opacity-50 w-full disabled:opacity-50 disabled:cursor-not-allowed"
        {...props}
      >
        {children}
      </button>
    );
  }

  return (
    <button
      className="bg-blue-500 text-white px-4 py-2 rounded-md transition-all duration-300 ease-in-out hover:bg-blue-600 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:ring-opacity-50 w-full disabled:opacity-50 disabled:cursor-not-allowed"
      {...props}
    >
      {children}
    </button>
  );
};

export default Button;
