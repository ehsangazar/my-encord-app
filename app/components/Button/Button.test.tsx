import renderer from "react-test-renderer";

import Button from "./Button";

it("Button component should render correctly", () => {
  const tree = renderer.create(<Button>Click me</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
