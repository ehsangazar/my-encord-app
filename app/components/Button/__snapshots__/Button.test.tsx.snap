// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly 1`] = `
<button
  className="bg-blue-500 text-white px-4 py-2 rounded-md transition-all duration-300 ease-in-out hover:bg-blue-600 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:ring-opacity-50 w-full disabled:opacity-50 disabled:cursor-not-allowed"
>
  Click me
</button>
`;
